### ISSUE TITLE
[device-port] [rolex] Xiaomi Redmi 4a

### ISSUE THREAD

**Issue by echosalik** [Halium/projectmanagement#92]
Tree: **halium-7.1**

- [x] Create manifest [xiaomi_rolex.xml](https://github.com/echosalik/halium-devices/blob/19d3b248d673c56a318e9ec0039cdb5ebb9493c6/manifests/xiaomi_rolex.xml)
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash
- [ ] libhybris tests
  * [ ] test_gps
  * [ ] test_hwcomposer
  * [x] test_lights
  * [x] test_vibrator
  * [x] test_wifi
  * [ ] test_sensors
  * [ ] test_audio
  * [ ] test_camera
  * [ ] test_input
  * [ ] test_nfc
  * [ ] test_recorder

Had to remove the camera folder in device tree **android_device_xiaomi_rolex**.
It was trying to find a static class: `android.hardware.camera.common@1.0-helper`
I tired to comment that out in the Android.mk of `QCamera2` folder but there was a lot of dependencies. I tired debugging it by manually adding files and changing class names to en-cooperate the name-spaces but it still gave errors. So I just removed the camera folder altogether which allowed build to complete.

EDIT:
had to add this: ``CONFIG_ANDROID_PARANOID_NETWORK=n`` for wifi

**Comment by ezs123** [comment]
Did you mannaged to develope a working ubports rom for rolex?
Can we test it?

**Comment by echosalik** [comment]
@ezs123 Nope. I failed with hwcomposer. I got a working terminal/ssh with Wireless but no GUI what so ever.
I am a newb when it comes to this. So if anyone who has better experience may have better luck than me.

**Comment by Foxtrot47** [comment]
From where did u get the kernel source code??.
Im stuck at building the kernel.
Something like I2C_TRANSFER undefined errors. Can u help me

**Comment by echosalik** [comment]
> From where did u get the kernel source code??.
> Im stuck at building the kernel.
> Something like I2C_TRANSFER undefined errors. Can u help me

Hey you can use the [source](https://github.com/echosalik/android_kernel_xiaomi_msm8917) on my profile or any of the following below:
https://github.com/ECr34T1v3/android_kernel_xiaomi_msm8917
https://github.com/redmidevs/android_kernel_xiaomi_msm8917
https://github.com/muralivijay/android_kernel_xiaomi_msm8917

Also I do remember I had this error, i guess I switched sources due to that error. My source is also based on one of the above (mostly probably muralivijay's source).

**Comment by Foxtrot47** [comment]
I used Najahi's(official Resurrection Remix maintainer for redmi 4a) sources . Since it was stable.
I will try it with his source. Please do upload your work.

**Comment by echosalik** [comment]
> I used Najahi's(official Resurrection Remix maintainer for redmi 4a) sources . Since it was stable.
> I will try it with his source. Please do upload your work.

My work is already on my profile, feel free to check it out.

**Comment by raananelefant** [comment]
Let me know if you need a tester.

**Comment by Foxtrot47** [comment]
I  was not working on this project. Been busy trying Android 10 ROM for 4a. But ..... Im gonna get back into it cause Android is boring.
@echosalik If ur interested I'll rent an Amazon cloud instance and try buding on it. But I forgot many of its setup . Would you kindly help me??

**Comment by echosalik** [comment]
I honestly don't remember anything myself. I have been busy with work so I haven't been able to try this out, also because this is my daily use phone. Plus I was waiting for hybris 8. 
I had building problems with 7.1 kernels, mostly because by the time xiaomi released it's kernel, android 8 was out and people started developing for android 8, which actually meant 7.1 kernels were kinda buggy and unfinished.
Also I am also on pixel experience 10, i like it. Not many bugs.

**Comment by Foxtrot47** [comment]
I'm too on Pixel XP 10.  Yeah there are lot of bugs.... So you're saying you won't return back. Ok then please close the thread if ur not building it

**Comment by echosalik** [comment]
I am not returning anytime back soon. So yeah i think i should close the issue, reopen once i have time for it.

**Comment by areyoudeveloper** [comment]
> I honestly don't remember anything myself. I have been busy with work so I haven't been able to try this out, also because this is my daily use phone. Plus I was waiting for hybris 8.
> I had building problems with 7.1 kernels, mostly because by the time xiaomi released it's kernel, android 8 was out and people started developing for android 8, which actually meant 7.1 kernels were kinda buggy and unfinished.
> Also I am also on pixel experience 10, i like it. Not many bugs.

hi im missing vendor rolex

**Comment by echosalik** [comment]
> 
> 
> > I honestly don't remember anything myself. I have been busy with work so I haven't been able to try this out, also because this is my daily use phone. Plus I was waiting for hybris 8.
> > I had building problems with 7.1 kernels, mostly because by the time xiaomi released it's kernel, android 8 was out and people started developing for android 8, which actually meant 7.1 kernels were kinda buggy and unfinished.
> > Also I am also on pixel experience 10, i like it. Not many bugs.
> 
> hi im missing vendor rolex

Try this out.
https://github.com/PixelExperience-Devices/device_xiaomi_rolex
You can search for vendors by searching on github. Some repos have been bought down by DCMA so you might have to dig around forums like XDA.

**Comment by areyoudeveloper** [comment]
> > > I honestly don't remember anything myself. I have been busy with work so I haven't been able to try this out, also because this is my daily use phone. Plus I was waiting for hybris 8.
> > > I had building problems with 7.1 kernels, mostly because by the time xiaomi released it's kernel, android 8 was out and people started developing for android 8, which actually meant 7.1 kernels were kinda buggy and unfinished.
> > > Also I am also on pixel experience 10, i like it. Not many bugs.
> > 
> > 
> > hi im missing vendor rolex
> 
> Try this out.
> https://github.com/PixelExperience-Devices/device_xiaomi_rolex
> You can search for vendors by searching on github. Some repos have been bought down by DCMA so you might have to dig around forums like XDA.
ninja: error: '/home/darkness/Documents/halium/out/host/linux-x86/framework/signapk.jar', needed by '/home/darkness/Documents/halium/out/target/product/rolex/obj/APPS/PowerOffAlarm_intermediates/package.apk', missing and no known rule to make it
build/core/ninja.mk:151: recipe for target 'ninja_wrapper' failed
make: *** [ninja_wrapper] Error 1
make: Leaving directory '/home/darkness/Documents/halium'

error from what?

**Comment by areyoudeveloper** [comment]
> > > > I honestly don't remember anything myself. I have been busy with work so I haven't been able to try this out, also because this is my daily use phone. Plus I was waiting for hybris 8.
> > > > I had building problems with 7.1 kernels, mostly because by the time xiaomi released it's kernel, android 8 was out and people started developing for android 8, which actually meant 7.1 kernels were kinda buggy and unfinished.
> > > > Also I am also on pixel experience 10, i like it. Not many bugs.
> > > 
> > > 
> > > hi im missing vendor rolex
> > 
> > 
> > Try this out.
> > https://github.com/PixelExperience-Devices/device_xiaomi_rolex
> > You can search for vendors by searching on github. Some repos have been bought down by DCMA so you might have to dig around forums like XDA.
> > ninja: error: '/home/darkness/Documents/halium/out/host/linux-x86/framework/signapk.jar', needed by '/home/darkness/Documents/halium/out/target/product/rolex/obj/APPS/PowerOffAlarm_intermediates/package.apk', missing and no known rule to make it
> > build/core/ninja.mk:151: recipe for target 'ninja_wrapper' failed
> > make: *** [ninja_wrapper] Error 1
> > make: Leaving directory '/home/darkness/Documents/halium'
> 
> error from what?

im use rolex vendor from muralivijay@github

**Comment by echosalik** [comment]
> 
> error from what?

~~What java version are you using?~~
Try this link:
http://docs.halium.org/en/latest/porting/common-system-build-errors.html

Also, try using the IRC channels, those might be more helpful.

**Comment by doniks** [comment]
@areyoudeveloper thanks for PR https://github.com/Halium/halium-devices/pull/203 I'm reopening this if you want to continue here. If you want to start over with a new one, feel free to close again
